src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -479,13 +502,9 @@ private static Expression AddResponseWritingToMethodCall(Expression methodCall,
 
             return async (target, httpContext) =>
             {
-                object? bodyValue;
+                object? bodyValue = defaultBodyValue;
 
-                if (factoryContext.AllowEmptyRequestBody && httpContext.Request.ContentLength == 0)
-                {
-                    bodyValue = defaultBodyValue;
-                }
-                else
+                if (httpContext.Request.ContentLength != 0 && httpContext.Request.HasJsonContentType())
```

Comment:
:+1: You're referring to `IHttpRequestBodyDetectionFeature.CanHaveBody` or the HttpRequest extension method `CanHaveBody()` added by https://github.com/dotnet/aspnetcore/pull/24984. Most of the time for the kind of POST or PUT requests this would be used for, it works out to the same thing (this is a nullable comparison so this branch should still be taken when ContentLength isn't set), but we might as well use the new feature.

Reply:
Done.